### PR TITLE
Fixes #9302: Time since last generation is limited to hours

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/9302

### CODE REVIEW

File: rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala [hunk 1]
```diff
@@ -64,12 +64,9 @@ object DateFormaterService {
...
-  def formatPeriod(duration:Duration) : String = {
-    if(duration.getMillis < 1000) "less than 1 s"
-    else periodFormatter.print(duration.toPeriod)
-  }
-
   def getFormatedPeriod(start : DateTime, end : DateTime) : String = {
-    formatPeriod(new Duration(start,end))
+    val duration =  new Duration(start,end)
+    if(duration.getMillis < 1000) "less than 1 s"
+    else periodFormatter.print(duration.toPeriodFrom(start))
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala]: in else do:  <pre>{
   val period = duration.toPeriod(PeriodType.YearMonthDayTime)
  periodFormatter.print(period)
}</pre>

Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala]: no, it should already be done, see the def of periodformatter l64 and after:
```
   private[this] val periodFormatter = new PeriodFormatterBuilder().
     appendDays().
     appendSuffix(" day", " days").
appendSeparator(", ")....
```

Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala]: In fact, I don't understand why days are not display now.

Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala]: The bug is in toPeriod, which create by default a periodType "Time" which is only "hours, minutes, seconds, millis"

Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala]: Hmm not sure in fact, documentation is not ... very clear

Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/DateFormaterService.scala]: So, after some error & try, the correct code is: 

```
import org.joda.time.chrono.ISOChronology
....
def formatPeriod(duration:Duration) : String = {
if(duration.getMillis < 1000) "less than 1 s"
else periodFormatter.print(duration.toPeriod(ISOChronology.getInstanceUTC))
}
```

**Comment by RaphaelGauthier** [comment]
Commit modified

**Pull request merged**